### PR TITLE
Validation rules for LRO PUTs

### PR DESCRIPTION
**Pull request by dsgouda**
address issue https://github.com/Azure/autorest/issues/1590

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/LROStatusCodesValidation.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace AutoRest.Swagger.Validation
+{
+    public class LROStatusCodesValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "blah";
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/LROStatusCodesValidation.cs]: please include an ID, just increase by 1 the last one used.

Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/LROStatusCodesValidation.cs]: could you also include documentation for this rule?

File: src/modeler/AutoRest.Swagger/Validation/LROStatusCodesValidation.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        /// </remarks>
+        public override string MessageTemplate => "200/201 Responses of long running operations must have a schema definition for return type. OperationId: '{0}', Response code: '{1}'";
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Warning;
+
+        /// <summary>
+        /// Verifies if a tracked resource has a corresponding ListBySubscription operation
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/LROStatusCodesValidation.cs]: please update comment

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs [hunk 3]
```diff
@@ -594,6 +594,13 @@ public void PutResponseResourceValidationTest()
         }
 
         [Fact]
+        public void LROStatusCodesValidationTest()
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: could run this rule on the repo and see how much it flags? It'd be good to know if the cases are real or if it starts flagging false positives.

File: Samples/2a-arm-validation/shell/stderr.txt [hunk 4]
```diff
@@ -26,6 +26,8 @@ ERROR (XmsExamplesProvidedValidation/M2022/SDKViolation): Please provide x-ms-ex
     - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-storage/2015-06-15/swagger/storage.json:362:8 ($.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/regenerateKey"].post.operationId)
 ERROR (XmsExamplesProvidedValidation/M2022/SDKViolation): Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operations. Operation: 'Usage_List'
     - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-storage/2015-06-15/swagger/storage.json:412:8 ($.paths["/subscriptions/{subscriptionId}/providers/Microsoft.Storage/usages"].get.operationId)
+WARNING (LROStatusCodesValidation/blah/SDKViolation): 200/201 Responses of long running operations must have a schema definition for return type. OperationId: 'StorageAccounts_Create', Response code: '200'
```
Comment (veronicagg) [inline Samples/2a-arm-validation/shell/stderr.txt]: i looked into this json file and I saw a ref for the operation https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2015-06-15/swagger/storage.json#L117, am I missing something? Not sure why the warning is reported.

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs (hunk 3 above)
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: Will do this and post the results

Comment (dsgouda) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: Ran linter rules against the whole repo, a cursory look at the logs did not reveal any false positives. It does flag some cases where it does make sense to not have a response model for 201/200 status. Eg: `Apis_CreateOrUpdate` or `FileSystem_RemoveDefaultAcl`

**Pull request merged**